EngineLayer/GlycoSearch/GlycoSearchEngine.cs

```diff
@@ -0,0 +1,609 @@
+
+
+            if (glycoSearchType == GlycoSearchType.OGlycanSearch)
+            {
+                GlycanBox.GlobalOGlycans = GlycanDatabase.LoadGlycan(GlobalVariables.OGlycanLocations.Where(p => System.IO.Path.GetFileName(p) == _oglycanDatabase).First(), true, true).ToArray();
+                GlycanBox.GlobalOGlycanModifications = GlycanBox.BuildGlobalOGlycanModifications(GlycanBox.GlobalOGlycans);
+                GlycanBox.OGlycanBoxes = GlycanBox.BuildOGlycanBoxes(_maxOGlycanNum, false).OrderBy(p => p.Mass).ToArray();
+            }
+            else if (glycoSearchType == GlycoSearchType.NGlycanSearch)
+            {
+                NGlycans = GlycanDatabase.LoadGlycan(GlobalVariables.NGlycanLocations.Where(p => System.IO.Path.GetFileName(p) == _nglycanDatabase).First(), true, false).OrderBy(p => p.Mass).ToArray();
+                //TO THINK: Glycan Decoy database.
+                //DecoyGlycans = Glycan.BuildTargetDecoyGlycans(NGlycans);
+            }
+            else if (glycoSearchType == GlycoSearchType.N_O_GlycanSearch)
+            {
+                GlycanBox.GlobalOGlycans = GlycanDatabase.LoadGlycan(GlobalVariables.OGlycanLocations.Where(p => System.IO.Path.GetFileName(p) == _oglycanDatabase).First(), true, true).ToArray();
+                GlycanBox.GlobalOGlycanModifications = GlycanBox.BuildGlobalOGlycanModifications(GlycanBox.GlobalOGlycans);
+                GlycanBox.OGlycanBoxes = GlycanBox.BuildOGlycanBoxes(_maxOGlycanNum, false).OrderBy(p => p.Mass).ToArray();
+
+                NGlycans = GlycanDatabase.LoadGlycan(GlobalVariables.NGlycanLocations.Where(p => System.IO.Path.GetFileName(p) == _nglycanDatabase).First(), true, false).OrderBy(p => p.Mass).ToArray();
+                //TO THINK: Glycan Decoy database.
+                //DecoyGlycans = Glycan.BuildTargetDecoyGlycans(NGlycans);
+            }
+
+        }
+
+        private Glycan[] NGlycans { get; }
+        //private Glycan[] DecoyGlycans { get; }
+
+        protected override MetaMorpheusEngineResults RunSpecific()
+        {
+            double progress = 0;
+            int oldPercentProgress = 0;
+            ReportProgress(new ProgressEventArgs(oldPercentProgress, "Performing crosslink search... " + CurrentPartition + "/" + CommonParameters.TotalPartitions, NestedIds));
+
+            byte byteScoreCutoff = (byte)CommonParameters.ScoreCutoff;
+
+            int maxThreadsPerFile = CommonParameters.MaxThreadsToUsePerFile;
+            int[] threads = Enumerable.Range(0, maxThreadsPerFile).ToArray();
+            Parallel.ForEach(threads, (scanIndex) =>
+            {
+                byte[] scoringTable = new byte[PeptideIndex.Count];
+                List<int> idsOfPeptidesPossiblyObserved = new List<int>();
+
+                byte[] secondScoringTable = new byte[PeptideIndex.Count];
+                List<int> childIdsOfPeptidesPossiblyObserved = new List<int>();
```

Comment:
unused variable
These three unused variables feel like something was supposed to happen with them

Reply:
Those will be used for N-glyco search. I will surely use them, but not now.